client/components/post-schedule/clock.jsx

```diff
@@ -81,6 +108,28 @@ class PostScheduleClock extends Component {
 	setTime = ( event, modifiers = this.getTimeValues() ) => {
 		const date = moment( this.props.date ).set( modifiers );
 		this.props.onChange( date, modifiers );
+	};
+
+	setAmPm( event, amOrPm ) {
+		this.refs.amPmReference.value = amOrPm;
+		this.setTime( event );
+	}
+
+	/**
+	 * Converts a 12-hour time to a 24-hour time, depending on time format.
+	 *
+	 * @param {Number}  hour The hour to convert.
+	 * @return {Number}      The converted hour.
+	 */
+	maybeConvertHour( hour ) {
```

Comment:
I really dislike the maybe naming pattern. Perhaps something a bit more verbose? `convert12HourTo24HourTime` and move the is12Hour check to the caller.